tests/models/llama/test_tokenization_llama.py

```diff
@@ -822,3 +825,26 @@ def test_special_tokens_strip(self):
         self.assertEqual(input_ids, [284, 1, 156])
         tokens = self.tokenizer.tokenize("No <s> ▁He")
         self.assertEqual(tokens, ["▁No", "<s>", "▁He"])  # spaces are eaten by rstrip / lstrip
+
+
+@require_read_token
+class TikTokenIntegrationTests(unittest.TestCase):
+    """
+    A class that regroups important test to make sure that we properly handle the special tokens.
+    """
+
+    def test_tiktoken_llama(self):
+        model_path = "hf-internal-testing/Llama3-Instruct-Internal"
+
+        tiktoken_tokenizer = AutoTokenizer.from_pretrained(model_path, legacy=False)
+        self.assertTrue(isinstance(tiktoken_tokenizer, PreTrainedTokenizerFast))
+        tokens = tiktoken_tokenizer.encode("Hey there")
+        self.assertEqual(tokens, [128000, 19182, 1070])
+
+        tmpdirname = tempfile.mkdtemp()
+        tiktoken_tokenizer.save_pretrained(tmpdirname)
+        tokenizer_reloaded = AutoTokenizer.from_pretrained(tmpdirname)
+        self.assertTrue(isinstance(tokenizer_reloaded, PreTrainedTokenizerFast))
+        tokens = tokenizer_reloaded.encode("Hey there")
+        self.assertEqual(tokens, [128000, 19182, 1070])
+        shutil.rmtree(tmpdirname)
```

Comment:
cool, can we go a bit deeper, use `PreTrainedTokenizerFast` for example, and make sure that loading using `from_slow` vs directly loading the already converted works?